python/Ganga/GPIDev/Base/VPrinter.py

```diff
@@ -120,7 +124,10 @@ def simpleAttribute(self, node, name, value, sequence):
             # else:
             #    print 'no transformation'
             if isclass(value):
-                print(self.indent(), name, '=', repr(value), end = '', file=self.out)
+                if self._interactive is True:
+                    print(self.indent(), name, '=', str(value), end = '', file=self.out)
+                else:
+                    print(self.indent(), name, '=', repr(value), end = '', file=self.out)
```

Comment:
Changing the behaviour of the jobtree etc to use a repr instead of a str changes the display behaviour of the job.
Before the ```repr``` function was used jobtree and subjobs would display correctly in the summary of a job. Now they display a really long line with formatting and ```\n``` characters and such. Using a string method to parse this gets back to the original intended behaviour.
Interactive matches the interactive mode of being on the IPython interactive prompt. We expect subtly different behaviour of some functions when we're using this for displaying the output. (Sure we could do some introspection and change the behaviour dynamically right at the repr/string level but this becomes overkill imo)
The default behaviour hasn't changed unless the VPrinter is called with pretty_repr for storing files to disk, in which case it's a simple matter to expand the API there to accept 1 additional boolean argument